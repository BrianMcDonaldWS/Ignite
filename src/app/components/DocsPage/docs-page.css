.contentArea {
  flex: 1;
  margin: auto;
  max-width: 1800px;
  width: 100%;
}

.content {
  width: 100%;
  margin: 0 auto 15px;
  padding-top: 0;
  padding-top: 3rem;
  padding-bottom: 5rem;

  details {
    padding-bottom: 20px;
  }

  @media screen and (min-width: 769px) {
    max-width: 950px;
  }
}

.Page {
  padding-left: 45px;
  padding-right: 45px;
}
