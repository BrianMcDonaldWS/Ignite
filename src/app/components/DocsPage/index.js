import React from 'react';
import PropTypes from 'prop-types';
import makeClass from 'classnames';
import ReactRouterPropTypes from 'react-router-prop-types';
import { Link } from 'react-router-dom';

import { default as Sidebar } from '../Sidebar';
import styles from './docs-page.css';

const DocsPage = ({ Page, SidebarComponent, location, plugins, className }) => (
  <div id="root" className={makeClass(styles.contentArea)}>
    <div className={makeClass(className, 'columns')}>
      <Sidebar
        className="column is-one-third-tablet is-one-quarter-desktop box"
        content={SidebarComponent}
        currentPage={`${location.pathname}${
          location.hash ? location.hash : ''
        }`}
      />

      <div
        className={makeClass(
          styles.content,
          'column',
          'content',
          'is-two-thirds-tablet',
          'is-three-quarters-desktop'
        )}
      >
        <Page plugins={plugins} className={styles.Page} />
      </div>
    </div>
  </div>
);

DocsPage.propTypes = {
  Page: PropTypes.node.isRequired,
  SidebarComponent: PropTypes.node.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  plugins: PropTypes.array,
  className: PropTypes.string
};

DocsPage.defaultProps = {
  plugins: [],
  className: null
};

export default DocsPage;
