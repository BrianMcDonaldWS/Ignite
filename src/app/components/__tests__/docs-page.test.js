import React from 'react';

import DocsPage from '../DocsPage';
import renderToJson from './utils/render-to-json';

test('DocsPage', () => {
  expect(
    renderToJson(<DocsPage Page={() => {}} location={{ pathname: '/' }} />)
  ).toMatchSnapshot();
});

test('DocsPage - with hash id', () => {
  expect(
    renderToJson(
      <DocsPage Page={() => {}} location={{ pathname: '/', hash: 'someId' }} />
    )
  ).toMatchSnapshot();
});
