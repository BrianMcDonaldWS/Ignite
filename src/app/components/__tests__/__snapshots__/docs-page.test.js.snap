// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocsPage - with hash id 1`] = `
<div
  className="contentArea"
  id="root"
>
  <div
    className="columns"
  >
    <Sidebar
      className="column is-one-third-tablet is-one-quarter-desktop box"
      content={null}
      currentPage="/someId"
    />
    <div
      className="content column content is-two-thirds-tablet is-three-quarters-desktop"
    >
      <Page
        className="Page"
        plugins={Array []}
      />
    </div>
  </div>
</div>
`;

exports[`DocsPage 1`] = `
<div
  className="contentArea"
  id="root"
>
  <div
    className="columns"
  >
    <Sidebar
      className="column is-one-third-tablet is-one-quarter-desktop box"
      content={null}
      currentPage="/"
    />
    <div
      className="content column content is-two-thirds-tablet is-three-quarters-desktop"
    >
      <Page
        className="Page"
        plugins={Array []}
      />
    </div>
  </div>
</div>
`;
